202208_galactic_foregrounds/PLAN.md

```diff
@@ -0,0 +1,81 @@
+Foreground simulation with realistic bandpasses and new Dust/Synchrotron models
+===============================================================================
+
+Produce full sky simulations of Galactic foregrounds using PySM 3 with the new Dust and Synchrotron models, using realistic bandpasses instead of top-hat.
+CMB and Extragalactic will be in a future release, [waiting for the new run of Websky](https://github.com/galsci/pysm/issues/120)
+
+## Foreground models
+
+The plan is to run 3 different sky models, optimistic/baseline/pessimistic, see the discussion in [this PySM 3 issue](https://github.com/galsci/pysm/issues/103#issuecomment-1081241879), for convenience here the 3 sets of models:
+
+1. d9,s4,f1,a1,co1
+2. d10,s5,f1,a1,co3
+3. d12,s7,f1,a2,co3
+
+Documentation reference:
+
+* `d9` `d10` GNILC based models and `d12` MKD 3D layered dust model: https://pysm3.readthedocs.io/en/latest/models.html#dust
+* Synchrotron models `s4` and `s5`: https://pysm3.readthedocs.io/en/latest/models.html#synchrotron
+* CO: https://pysm3.readthedocs.io/en/latest/models.html#co-line-emission
+* All other models are the same of PySM 2: https://pysm3.readthedocs.io/en/latest/models.html
+
+There are 2 issues:
+
+### Rotation to Equatorial
+
+In previous runs I had created Equatorial versions of all the templates to avoid having to rotate the maps because it was not supported by libsharp, for this release we can stop using libsharp and instead use all the Galactic templates and then rotate to Equatorial at the last step when we are in spherical harmonics space for smoothing.
+
+### Low resolution models
+
+* d9, d10, s4, s5 and s7 are available up to 8192
+* d12 to 2048
+* f1, a1, a2 to 512, however we have an extrapolation to 4096 by @NicolettaK, see <https://portal.nersc.gov/project/cmb/so_pysm_models_data/>
+* co? to 2048
```

Comment:
unfortunately we don't have high resolution template for co at nside=2048.We could perhaps inject small scales as we do w/ dust, but currently we don't have such a model...

Reply:
@giuspugl we do have CO maps at Nside 2048, 

https://portal.nersc.gov/project/cmb/pysm-data/co/

are you saying they are just Nside 512 maps repixelized at 2048?

Reply:
yes ! the planck maps  (type 1)are produced at nside=2048 but they are too noisy to be used as templates for CO . I therefore smoothed maps at 1deg reso and degraded to nside=512. 
see more details here 
https://giuspugl.github.io/reports/Adding_CO_to_pysm

Reply:
ok, then we should use 512,
also added link to your post on the docs:

https://github.com/galsci/pysm/commit/f4fd10fb1bc14d3cc7eca5ae90fc588d9cea739a